### ISSUE TITLE
Support for alternative Jupyter kernel

### ISSUE THREAD

**Issue by BoPeng** [spyder-ide/spyder#3314]
We have been using spyder as an IDE for a workflow engine that is based on Python and Jupyter (https://github.com/BoPeng/SOS). Spyder is almost perfect for this job because of its support for cells (natural counterpart for workflow steps), console (for interactive use), editor (for the development of script) and object inspector. 

Because spyder does not support alternative kernels, right now we have to start a SoS kernel and connect to it in spyder using "Consoles -> Connecting to an existing kernel". Is it possible for spyder to provide a commnad line option to use an alternative kernel (`--kernel`) or connect to an existing kernel (`--connect`) when spyder starts?

In addition, we currently patch `spyderlib.utils.sourcecode` and `spyderlib.config` to let spyder recognize files with extensions `.sos` and `.sosnb`, would it be possible to config spyder to access other file extensions (through spyder configuration file) or add these extensions to spyder?

I can certainly provide patches for these changes if they are considered acceptable.

**Comment by ccordoba12** [comment]
We've been talking with @goanpeca about this. Unfortunately we're not ready for it but we're almost there :-)

I think after we release 3.0, we'll start to think on how to support alternative kernels.

**Comment by goanpeca** [comment]
Yes :-) @BoPeng definitely something that we are working into for a future release

**Comment by BoPeng** [comment]
Thanks, this is reassuring. At that time SoS should also be more mature and work better with Spyder.

**Comment by ccordoba12** [comment]
We're working to support different kernels in Spyder 4.0 :-)

**Comment by BoPeng** [comment]
BTW, [SoS](https://github.com/BoPeng/SOS) provides a script [patch_spyder.py](https://github.com/BoPeng/SOS/blob/master/misc/patch_spyder.py) to monkey-patch spyder so that it accepts `.sos` as its source file, and can start with a `sos` kernel using command line option `--kernel sos`. This is by no means a proper implementation but at least allows spyder to work with SoS before a proper solution is provided.
